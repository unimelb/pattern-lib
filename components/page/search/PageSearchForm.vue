<template>
  <form action="https://search.unimelb.edu.au" method="get" class="page-header-search__form">
    <input ref="input" class="page-header-search__input" name="q" type="search" placeholder="Search the University" aria-label="Search the University"><button class="page-header-search__submit" type="submit">
      <SvgIcon class="page-header-search__icon" name="search" />
      <span class="screenreaders-only">Go</span>
    </button>
    <slot></slot>
  </form>
</template>

<script>
export default {
  name: 'page-search-form',
};
</script>
