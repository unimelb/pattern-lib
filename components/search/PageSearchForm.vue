<template>
  <form
    action="https://search.unimelb.edu.au"
    method="get"
    class="page-header-search__form">
    <input
      ref="input"
      class="page-header-search__input"
      name="q"
      type="search"
      placeholder="Search the University"
      aria-label="Search the University of Melbourne"><button
        class="page-header-search__submit"
        type="submit">
        <SvgIcon
          class="page-header-search__icon"
          name="search" />
        <span class="screenreaders-only">Go</span>
      </button>
    <slot />
  </form>
</template>

<script>
import SvgIcon from '../icons/SvgIcon.vue';

export default {
  name: 'PageSearchForm',
  components: { SvgIcon },
};
</script>
