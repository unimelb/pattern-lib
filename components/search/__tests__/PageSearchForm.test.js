import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import PageSearchForm from '../PageSearchForm.vue';

expect.extend(toHaveNoViolations);

describe('PageSearchForm', () => {
  it('should match snapshot', () => {
    const result = shallow(PageSearchForm).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(PageSearchForm).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
