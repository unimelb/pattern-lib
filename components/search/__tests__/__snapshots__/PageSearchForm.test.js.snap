// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSearchForm should match snapshot 1`] = `
<form
  action="https://search.unimelb.edu.au"
  class="page-header-search__form"
  method="get"
>
  <input
    aria-label="Search the University of Melbourne"
    class="page-header-search__input"
    name="q"
    placeholder="Search the University"
    type="search"
  />
  <button
    class="page-header-search__submit"
    type="submit"
  >
    <!---->
     
    <span
      class="screenreaders-only"
    >
      Go
    </span>
  </button>
   
</form>
`;
